ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java

```diff
@@ -50,6 +51,10 @@
   private DbEndpoint connectedEndpoint;
   private List<DbEndpoint> sortedDbEndpoints;
   private EndpointComparator endpointComparator;
+  // Time of most recent attempt to connect to optimal endpoint
+  private long lastConnectionAttemptTime = 0;
+  // TODO: make config property
+  private long connectionRetryWaitTime = TimeUnit.SECONDS.toMillis(30);
```

Comment:
LGTM, but can we make this configurable? and if it's 0 or -1, maybe don't retry it.

Reply:
Yes, I will add that in a separate PR after Arun's transaction changes are merged (want to minimize conflicts since a few classes need to change to pass in the config).